### DISCUSSION TITLE
How to depend on 'detektMain' task?

### DISCUSSION THREAD

**Discussion by Den-Rimus** [detekt/detekt#3428] [Q&A] [answered]
I need to set up a chain of tasks, some of which will run after `detektMain`

I am able to run `./gradlew detektMain` in root directory of my android library project with multiple modules.
But trying to depend on it in root build.gradle.kts code will result in
```
A problem occurred configuring root project 'my-awesome-project'.
> Task with name 'detektMain' not found in root project 'my-awesome-project'.
```

I looked up setting up these tasks (like `detektMain` or `detektTest`) in detekt source code and figured that registering them is made in `project.afterEvaluate {` closure (file `DetektAndroid.kt:70`). Ok. I tried putting my `dependsOn` logic to such closure (which seems expected thing to do in this case, no?) but still no luck.

Can someone help me author these gradle tasks correctly?

**Comment by picklebento** [comment, marked as answer]
Thank you for raising the question: The discussion of android gradle tasks can also be seen in #3122 and #3240.

I have prototype #3334 to register android source set the same way as we register JVM source set, and does not use `afterEvaluate` block to enforce ordering of the execution in gradle.
This prototype caused us to rethink gradle tasks #3364.

Feel free to use the method in #3334 if this is an urgent task for you and I am available on #kotlinlang slack. The prototype in #3334 is based on the custom solution I built at my company to register Android tasks correctly.

We are probably going to evaluate different options in the next release after 1.16, and I would like to wait for AGP 4.2 to be fully released as well.
